lms/djangoapps/student_account/views.py

```diff
@@ -67,6 +67,10 @@ def login_and_registration_form(request, initial_mode="login"):
         initial_mode (string): Either "login" or "register".
 
     """
+    if initial_mode == "register" and \
+            'ALLOW_PUBLIC_ACCOUNT_CREATION' in settings.FEATURES and \
+            settings.FEATURES["ALLOW_PUBLIC_ACCOUNT_CREATION"]:
+        return HttpResponseNotFound('<h1>Page not found</h1>')
```

Comment:
You can use compact variant raise Http404

Reply:
Done